tools/src/main/java/com/nedap/archie/flattener/Flattener.java

```diff
@@ -610,31 +633,161 @@ private void flattenAttribute(CComplexObject root, CAttribute parent, CAttribute
             parent.setCardinality(getPossiblyOverridenValue(parent.getCardinality(), specialized.getCardinality()));
 
             if (specialized.getChildren().size() > 0 && specialized.getChildren().get(0) instanceof CPrimitiveObject) {
-                //TODO: is this correct? replace all specialized nodes
+                //in case of a primitive object, just replace all nodes
                 parent.setChildren(specialized.getChildren());
             } else {
 
-                Set<CObject> newChildObjects = new LinkedHashSet<>(specialized.getChildren());
-                for (CObject parentCObject : new ArrayList<>(parent.getChildren())) {
-                    List<CObject> matchingChildren = new ArrayList<>();
-                    for (CObject specializedChildCObject : specialized.getChildren()) {
+                //ordering the children correctly is tricky.
```

Comment:
It would be more readable to move this sibling reorder code to a seperate method. This way the steps taken to flatten an attribute are grouped together in flattenAttribute, and the code to reorder the siblings can be isolated

Reply:
Made much more readable in #7 , but exactly not in the way you describe here :)